arcgis-ios-sdk-samples/Content Display Logic/Model/SearchEngine.swift

```diff
@@ -14,165 +14,180 @@
 
 import UIKit
 
-class SearchEngine: NSObject {
-
-    static private let singleton = SearchEngine()
+class SearchEngine {
     
-    private var indexArray:[String]!
-    private var wordsDictionary:[String: [String]]!
-    private var isLoading = false
+    private var displayNamesByReadmeWords:[String: [String]] = [:]
+    private var isLoadingReadmeIndex = false
     
-    override init() {
-        super.init()
-        
-        self.commonInit()
-    }
+    let samples:[Sample]
```

Comment:
I'm just spitballing here, but it might make more sense to store the samples in a dictionary keyed off a unique value, say a UUID. The initializer could take any generic sequence of samples and then use that sequence to construct a dictionary. That dictionary could then be used for efficient lookup, rather than storing names and using those for lookup.

Reply:
That could be a better solution. How would you suggest constructing UUIDs for the samples?

Reply:
Something like this should do the trick:
```swift
self.samples = Dictionary(uniqueKeysWithValues: samples.lazy.map { (UUID(), $0) })
```